### DISCUSSION TITLE
Could we get support for ignoring builds rather than only aborting them?

### DISCUSSION THREAD

**Discussion by joostdecock** [vercel/community#4552] [Feature Question]
### Question

Hi there, I would like to make a suggestion for what seems like some real low-hanging fruit for optimization: The ability to ignore builds.

Now before you say *but there is the ignored build step thingie* hear me out:

The *Ignored Build Step* is a nice way to use fine-grained control over whether you want to build or not, but it sits too late in the pipeline to really call it *ignore* anything. It really is an **Abort Build** rather than **Ignore Build**.

For example: we use an external integration (crowdin.com) for translation which pushes a flurry of commits to our `i18n` branch.
We never ever want to build this branch, and we have configured the script to prevent that from happening.
Still, all those builds are started, then cancelled:

```
[16:54:13.981] Running build in Washington, D.C., USA (East) – iad1 (Hive)
[16:54:14.089] Cloning github.com/freesewing/freesewing (Branch: i18n, Commit: 42d678e)
[16:54:27.268] Cloning completed: 13.179s
[16:54:27.269] Running "node ./skip_build.mjs"
[16:54:27.345] FreeSewing skip build check:
[16:54:27.345]   branch: i18n
[16:54:27.346]   commit: 42d678ef1b6667c46ca0443da8970f4b4df01d08
[16:54:27.346]   author: joostdecock
[16:54:27.346]   commit message: New translations en.md (German)
[16:54:27.346] skip-build
[16:54:27.346] 
[16:54:27.346] 🛑  Not building: Branch is included in branches to never build
[16:54:27.396] The Deployment has been canceled as a result of running the command defined in the "Ignored Build Step" setting.
```

Furthermore, this causes rate-limiting to kick in:

```
Too many requests - try again in 4 minutes (more than 60, code: "api-deployments-flood").
```

This seems like real low-hanging fruit for optimization. Allowing users to _truly_ ignore builds based on criteria that do not require the repo to be fetched (such as branch name) could save a ton of time and processing.

So I would like to humbly request this feature to be considered for implementation  🙏

**Comment by amyegan** [comment]
There's already a solution that seems perfect for your situation! 😄 

You can turn auto-deployments off with [`git.deploymentEnabled`](https://vercel.com/docs/projects/project-configuration/git-configuration#git.deploymentenabled) in a `vercel.json` file at the root of the repo. In your case, something like this should do the trick:
```json
{
  "git": {
    "deploymentEnabled": {
      "i18n": false
    }
  }
}
```

https://vercel.com/docs/projects/project-configuration/git-configuration#git.deploymentenabled

**Reply by joostdecock** [reply]
Please tell me this is relatively new, because the alternative is not a great look or me :see_no_evil: 

Thanks a lot, this is indeed perfect for my use-case :pray:

**Reply by amyegan** [reply]
It's fairly new and not one of our more flashy features. Totally understandable that you could miss it!

I'm glad it worked for you 🙂

**Reply by mmachatschek** [reply]
this is one of the best features of the last 6-12 months released for the vercel github integration 👍 stumbled over this just now 🙄
